### DISCUSSION TITLE
Add support for MyRadioOnline suite of websites

### DISCUSSION THREAD

**Discussion by tepiloxtl** [junguler/m3u-radio-music-playlists#36] [General]
There is a whole suite of websites under MyRadioOnline umbrella (MyOnlineRadio.de UKRadioLive.com MyRadioOnline.es MyRadioOnline.it MyOnlineRadio.at MyOnlineRadio.sk MyOnlineRadio.hu MyRadioEnDirect.fr MyOnlineRadio.nl IrishRadioLive.com MyRadioEnVivo.mx MyRadioOnline.ro MyRadioOnline.cl MyRadioEnVivo.ar). Notably, MyRadioOnline.net is NOT a part of the same network, though it seems to emulate the same look of other sites.

While many websites exists, data from all supported countries can be scraped from a single one, no matter which one. You'll probably want to look into /get-radio-list endpoint to see how it works.

I'm making my own scraper for the website in python. I don't know much about posix shell, but I can help with website itself

**Comment by junguler** [comment]
hello, i've aware of those websites, i wanted to scrape them but the issue is there is no genre specified for them, there are just a bunch of radio streams listed there in no order other than just popularity

as for the https://myradioonline.net/ it looks promising altho there is just a handful of genres to scrape from and overall there aren't that many radio streams listed, 7 pages of 20-30 streams per page, i'm not against the idea of adding this website however and will add it to the list

the next website i'll be scraping is already decided but i was busy updating the +checked+ folder this past few weeks so be on the lookout for that whenever it finally arrives

**Comment by junguler** [comment]
i'm going to convert this issue to a discussion, feel free to keep the conversation going if you want

**Reply by afkcodes** [reply]
can you list the next batch of the websites ? Pretty curious :)

**Reply by junguler** [reply]
i would as soon as i have it, there is not an actual list i'm referencing, i just randomly open a m3u playlist and find a radio station, search for "listen to station_name" and go thru the list of websites, i have mental list of the websites i've already checked/scraped so when new ones pop up i'll know

search engine diversity helps a lot too, i've stopped using google for searching completely, bing gives me a lot more new websites i wasn't seeing before, it doesn't seem to censor the results nearly as much

**Comment by tepiloxtl** [comment]
I went around and uploaded my scraper to github now, for anyone interested: https://github.com/tepiloxtl/nd-stuff/tree/main/mro-to-m3u

**Comment by afkcodes** [comment]
@junguler  can we also get the radios from [surfmusic.de](https://www.surfmusic.de/) is it possible ?

**Reply by junguler** [reply]
this website you linked is all over the place, there are some links that go to external websites and some that will open in the same website but in a pop-up window

it seems the ones that are stored locally in the website are easy to scrape, it's just that i'm in the middle of scraping another website right now and it might take a while to be completed

in short, yes, i will scrape it in the future but not everything listed there will be included

**Reply by afkcodes** [reply]
if that's the case i would recommend not to scrape it. i also had a look at it most of the radio open in new window. lets drop it.

btw which site are you now scraping. curious i am 😬

**Reply by junguler** [reply]
ok, yeah

this is the website i'm scraping https://liveradio24.com/ you can take a look at the temp repo https://github.com/junguler/m3u-temp i'm in the middle of it actually
